CheckoutKata/Checkout.cs

```diff
@@ -3,18 +3,41 @@ namespace CheckoutKata;
 public class Checkout
 {
     private readonly ISkuRetrievalService _skuRetrievalService;
+    private readonly ISkuSpecialRetrievalService _skuSpecialRetrievalService;
     
-    public Checkout(ISkuRetrievalService skuRetrievalService)
+    public Checkout(ISkuRetrievalService skuRetrievalService, ISkuSpecialRetrievalService skuSpecialRetrievalService)
     {
         _skuRetrievalService = skuRetrievalService;
+        _skuSpecialRetrievalService = skuSpecialRetrievalService;
     }
     
     public int Total { get; private set; }
+    
+    private readonly List<string> _skuCodes = new();
 
     public void Scan(string code)
     {
         var validSku = _skuRetrievalService.GetSkuByCode(code);
+        
+        _skuCodes.Add(validSku.Code);
+        
+        Console.WriteLine(_skuCodes);
+
+        
+        var skuSpecial = _skuSpecialRetrievalService.GetSkuSpecialByCode(code);
+        
+        Console.WriteLine(skuSpecial);
+
+        var count = _skuCodes.Count(skuCode => skuCode.StartsWith(code));
```

Comment:
Be careful with startswith - if you had a SkuCode "Apple" and code was "A" it would pass and you would not want it to

This is distinct behavior you could protect against in a test